adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/Features.scala

```diff
@@ -0,0 +1,238 @@
+    attributes.foreach(entry =>
+      entry._1 match {
+        // reserved keys in GFF3 specification
+        case "ID"            => f.setFeatureId(entry._2)
+        case "Name"          => f.setName(entry._2)
+        case "Target"        => f.setTarget(entry._2)
+        case "Gap"           => f.setGap(entry._2)
+        case "Derives_from"  => f.setDerivesFrom(entry._2)
+        case "Is_circular"   => f.setIsCircular(entry._2.toBoolean)
+        case "Alias"         => aliases += entry._2
+        case "Note"          => notes += entry._2
+        case "Parent"        => parentIds += entry._2
+        case "Dbxref"        => toDbxref(entry._2).foreach(dbxrefs += _)
+        case "Ontology_term" => toOntologyTerm(entry._2).foreach(ontologyTerms += _)
+        // commonly used keys in GTF/GFF2, e.g. via Ensembl
+        case "gene_id"       => f.setGeneId(entry._2)
+        case "transcript_id" => f.setTranscriptId(entry._2)
+        case "exon_id"       => f.setExonId(entry._2)
+        // unrecognized key, save to attributes
+        case _               => remaining += entry
+      }
+    )
+
+    // set list fields if non-empty
+    if (!aliases.isEmpty) {
+      f.setAliases(aliases)
+    }
+    if (!notes.isEmpty) {
+      f.setNotes(notes)
+    }
+    if (!parentIds.isEmpty) {
+      f.setParentIds(parentIds)
+    }
+    if (!dbxrefs.isEmpty) {
+      f.setDbxrefs(dbxrefs)
+    }
+    if (!ontologyTerms.isEmpty) {
+      f.setOntologyTerms(ontologyTerms)
+    }
+
+    // set remaining attributes if non-empty;
+    // any duplicate keys are lost at this point, last one in wins
+    if (!remaining.isEmpty) {
+      f.setAttributes(remaining)
+    }
+
+    f
+  }
+
+  /**
+   * Gather values from various feature fields into a sequence of attribute key value pairs.
+   *
+   * @param feature feature to gather values from
+   * @return values from various feature fields gathered into a sequence of attribute key value pairs
+   */
+  def gatherAttributes(feature: Feature): Seq[(String, String)] = {
+    val attrs = new ArrayBuffer[(String, String)]
+
+    // not sure why scala compiler requires this
+    def addBooleanTuple(b: java.lang.Boolean) = {
+      attrs += Tuple2("Is_circular", b.toString)
+    }
+
+    // ... and why Tuple2 is required here
+    Option(feature.getFeatureId).foreach(attrs += Tuple2("ID", _))
+    Option(feature.getName).foreach(attrs += Tuple2("Name", _))
+    Option(feature.getTarget).foreach(attrs += Tuple2("Target", _))
+    Option(feature.getGap).foreach(attrs += Tuple2("Gap", _))
+    Option(feature.getDerivesFrom).foreach(attrs += Tuple2("Derives_from", _))
+    Option(feature.getIsCircular).foreach(addBooleanTuple)
+    Option(feature.getGeneId).foreach(attrs += Tuple2("gene_id", _))
+    Option(feature.getTranscriptId).foreach(attrs += Tuple2("transcript_id", _))
+    Option(feature.getExonId).foreach(attrs += Tuple2("exon_id", _))
+    for (alias <- feature.getAliases) attrs += Tuple2("Alias", alias)
```

Comment:
I don't think that worked before, maybe for the same anon argument issue you mention above.  Will give it a try.